src/clients/go/pkg/types/main.go

```diff
@@ -73,12 +83,73 @@ func HexStringToUint128(value string) (Uint128, error) {
 
 // BigIntToUint128 converts a [math/big.Int] to a Uint128.
 func BigIntToUint128(value big.Int) Uint128 {
+	// big.Int bytes are big-endian so convert them to little-endian for Uint128 bytes.
 	bytes := value.Bytes()
-	return BytesToUint128(*(*[16]byte)(bytes))
+	swapEndian(bytes[:])
+	
+	// Only cast slice to bytes when theres enough.
+	if len(bytes) >= 16 {
+		return BytesToUint128(*(*[16]byte)(bytes))
+	}
+	
+	var padded [16]byte
+	copy(padded[:], bytes)
+	return BytesToUint128(padded)
 }
 
 // ToUint128 converts a integer to a Uint128.
 func ToUint128(value uint64) Uint128 {
 	values := [2]uint64{value, 0}
 	return *(*Uint128)(unsafe.Pointer(&values[0]))
 }
+
+var idLastTimestamp int64
+var idLastRandom [10]byte
+var idMutex sync.Mutex
+
+// Generates a Universally Unique and Sortable Identifier encoded as a Uint128. 
+func ID() Uint128 {
```

Comment:
Let's call it `CreateID()` and standardize the same docs comments across all languages.
It's nice to link the ULID spec as well.